persistence/file/src/main/java/de/fraunhofer/iosb/ilt/faaast/service/persistence/file/PersistenceFile.java

```diff
@@ -0,0 +1,136 @@
+import de.fraunhofer.iosb.ilt.faaast.service.persistence.file.util.FileHelper;
+import de.fraunhofer.iosb.ilt.faaast.service.util.AASEnvironmentHelper;
+import de.fraunhofer.iosb.ilt.faaast.service.util.DeepCopyHelper;
+import io.adminshell.aas.v3.dataformat.DeserializationException;
+import io.adminshell.aas.v3.model.Identifiable;
+import io.adminshell.aas.v3.model.Identifier;
+import io.adminshell.aas.v3.model.Reference;
+import io.adminshell.aas.v3.model.SubmodelElement;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Path;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * Implementation of
+ * {@link de.fraunhofer.iosb.ilt.faaast.service.persistence.Persistence} for a file storage.
+ * <p>
+ * Following types are not supported in the
+ * current version:
+ * <ul>
+ * <li>AASX packages
+ * <li>Package Descriptors
+ * <li>SubmodelElementStructs
+ * </ul>
+ */
+public class PersistenceFile extends PersistenceBasic<PersistenceFileConfig> {
+
+    private FileHelper fileHelper;
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(PersistenceFile.class);
+
+    @Override
+    public void initAASEnvironment(PersistenceFileConfig config) {
+        fileHelper = new FileHelper(config);
+        try {
+            if (!config.isLoadOriginalFileOnStartUp()) {
+                aasEnvironment = fileHelper.loadAASEnvironment();
+                Path filePath = fileHelper.getFilePath().toAbsolutePath();
+                if (aasEnvironment != null) {
+                    LOGGER.info("File Persistence uses existing model file {}", filePath);
+                }
+                else {
+                    LOGGER.info("File Persistence creates model file {}", filePath);
+                }
+            }
+            if (aasEnvironment == null) {
+                if (config.getEnvironment() != null) {
+                    aasEnvironment = config.isDecoupleEnvironment() ? DeepCopyHelper.deepCopy(config.getEnvironment()) : config.getEnvironment();
+                }
+                else {
+                    aasEnvironment = AASEnvironmentHelper.fromFile(new File(config.getModelPath()));
+                }
+                fileHelper.save(aasEnvironment);
+            }
+        }
+        catch (DeserializationException | IOException e) {
+            throw new IllegalArgumentException("Error deserializing AAS Environment", e);
+        }
+        identifiablePersistenceManager.setAasEnvironment(aasEnvironment);
+        referablePersistenceManager.setAasEnvironment(aasEnvironment);
+        packagePersistenceManager.setAasEnvironment(aasEnvironment);
+    }
+
+
+    /**
+     * Get the current file path of the model file used by the file persistence
+     *
+     * @return file path of the model file
+     */
+    public Path getFilePath() {
+        return fileHelper.getFilePath();
+    }
+
+
+    @Override
+    public void afterInit() {
+        fileHelper.save(super.getEnvironment());
```

Comment:
"super." should be redundant, make even make environment protected and access it directly